crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1165,6 +1165,7 @@ impl<TX: DbTx + 'static, N: NodeTypes> BlockNumReader for DatabaseProvider<TX, N
 
 impl<TX: DbTx + 'static, N: NodeTypesForProvider> BlockReader for DatabaseProvider<TX, N> {
     type Block = BlockTy<N>;
+    type BlockBody = BodyTy<N>;
```

Comment:
this type can be obtained from the Block type directly because this satisfies trait Block

https://github.com/paradigmxyz/reth/blob/1effb1952c58acb7435fb4ea2257e47ed868ab83/crates/storage/storage-api/src/block.rs#L59-L63